src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1155,21 +1155,33 @@ export default class DefaultDeviceController
       return true;
     }
     const deviceIds = DefaultDeviceController.getIntrinsicDeviceId(device);
+    this.logger.debug(
+      `Checking deviceIds ${deviceIds} of type ${typeof deviceIds} with groupId ${groupId}`
+    );
     if (typeof deviceIds === 'string' && groupId === this.getGroupIdFromDeviceId(kind, deviceIds)) {
       return true;
     }
     return false;
   }
 
   private getGroupIdFromDeviceId(kind: string, deviceId: string): string {
+    /* istanbul ignore else */
```

Comment:
Can this one cover the missing else in if condition at line 1173?

Reply:
We could not write unit test for this. The only time this triggers is when the infinite looping happens for audio input.

Reply:
Have we not seen this issue occur for the video devices as well, or is it just customers typically set audio first and thus encounter it first

Reply:
So the issue right now only happens to audio input because after select audio input, we call `restartLocalAudio` https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L1513 which will try to call choose audio input device again, expecting if we can reuse the exsting audio device. However, since that matching fails, it will try to choose the audio input device again -> Causing the infinite loop.